### PR TITLE
Added 401, 418, and 503 Errors

### PR DESCRIPTION
**Pull request by doamatto**
- Added 401, 418, and 503 Error Pages
- Adds in "Story Game" through error codes for the adventurous who either want to break to the end or want to start at the.. well.. start

- PR for Google Code-In -

### CODE REVIEW

**Review by saurabhshri — CHANGES_REQUESTED** [review]
Thank you for the PR! :) 

Let's review. For starters, the error handlers are missing. http://flask.pocoo.org/docs/1.0/patterns/errorpages/#error-handlers

**Comment by doamatto** [comment]
I just pushed with addition of error handlers. If that's all that's needed, I'll resubmit via GCI

**Comment by saurabhshri** [comment]
@Incrested Could you please add some tests and post screenshots/gif of the changes?

**Comment by doamatto** [comment]
I would if I know how to go to the error page. Am I supposed to cause the error or? Sorry: not positive on how to do this. Regardless, screenshots should be pretty to get you after a quick point in the right direction.

File: run.py [hunk 1]
```diff
@@ -253,6 +261,15 @@ def internal_server_error_handler(e):
         layout = LayoutHelper(logged_in=False)
     return render_template('500.html', layout=layout.get_entries()), 500
 
+@app.errorhandler(503)
+def unauthorized_handler(e):
```
Comment (saurabhshri) [inline run.py]: This should not be `unauthorized_handler`. 503 is `Service Unavailable`.

**Review by saurabhshri — CHANGES_REQUESTED** [review]
@Incrested To test if your pages are being served fine, you may create a test route and then render your error pages.

E.g.

```
@app.route('/401')
def errorrs():
    if g.user is not None:
        details = DetailsForTemplate(g.user.id, admin_dashboard=False)
        layout = LayoutHelper(logged_in=True, details=details)
    else:
        layout = LayoutHelper(logged_in=False)
    return render_template('401.html', layout=layout.get_entries()), 401
```

Then visit `{address}/401` to see it.

File: run.py (hunk 1 above)
Comment (doamatto) [reply run.py]: Just fixed it in 5d60b74

**Comment by doamatto** [comment]
I tried what you suggested @saurabhshri, although it seems it can never find the file that is mentioned (just responds when getting "/401" with a 404). Any ideas on if it's human error or if I might be about cc-web?

**Comment by saurabhshri** [comment]
@Incrested Please post your code, logs and screenshot so that we could find out :)

**Comment by doamatto** [comment]
Code can be found in the fork
Log/Screenshot: [Recorded Preview of Terminal](https://youtu.be/4NJEcTyw4BI)

Let me know if you need anything else :)

File: run.py [hunk 2]
```diff
@@ -253,6 +261,43 @@ def internal_server_error_handler(e):
...
+        layout = LayoutHelper(logged_in=True, details=details)
+    else:
+        layout = LayoutHelper(logged_in=False)
+    return render_template('503.html', layout=layout.get_entries()), 503
+
 if __name__ == '__main__':
     log.debug("Running App.")
     app.run(host='0.0.0.0')
+
+@app.route('/401')
```
Comment (saurabhshri) [inline run.py:277]: Please remove all these routes, I recommended it to you for testing in your local set-up. Also, looks like all of your routes have same function name, that can not be correct?

**Comment by saurabhshri** [comment]
Hi, are you still working on this?

**Comment by doamatto** [comment]
No: I'll reopen if I get the chance to do more in the future.

**Pull request closed without merging**